include/hexahedron_element.tcc

```diff
@@ -563,4 +563,68 @@ inline Eigen::MatrixXi
   //clang-format on
   return indices;
 }
+
+
+//!            Behind: F4
+//!        3      F2       2
+//!          0_ _ _ _ _ _0
+//!         /|           /|
+//!        / |          / |
+//!     7 0_ |_ _ _ _ _0 6|
+//!       |  |         |  |
+//!    F3 |  |         |  |   F1
+//!       |  0_ _ _ _ _|_ 0
+//!       | / 0        | / 1
+//!       |/     F0    |/
+//!       0_ _ _ _ _ _ 0
+//!     4               5  
+//!         Front: F5 
+//!
+//!
+//! Bottom face: F0, Right face: F1, Top face: F2, 
+//! Left face: F3, Behind face: F4, Front face: F5
+//! Return indices of a face of the element
+//! 8-noded hexahedron
+template <>
+inline Eigen::MatrixXi
+    mpm::HexahedronElement<3, 8>::face_indices(unsigned face_id) const {
+  
+  // Make return_indices
+  Eigen::Matrix<int, 4, 1> return_indices;
+  try {
+    // Check if face_id is within range
+    if (face_id < face_indices_hexahedron_.size()) {
+      return_indices = face_indices_hexahedron_.at(face_id).head(4);
```

Comment:
Since you are chopping, you might as well define a const variable locally for each template specialisation and return the correct size. Create local variables of face indices for different element types (template specailsation).

Reply:
Understood. I agree it's better. No private member then

Reply:
You are incurring more cost of creating another Eigen::Matrix by chopping, it would be more expensive than a function that has const variable returned each time. I don't understand what you mean by a library to store all data?

Reply:
Say for 2D quadrilateral 4-node, we need to know that for face 0, the associated nodes are 0 and 1. Then for face 1, the associated nodes are 1 and 2. What I meant by library is the data that map face id to the associated nodes. 

The function has input `face_id`. If we make a local variable, we need to make the 'library' every time and only use one in the end. 

I think having a private member is better. Yes, truncation of matrix has cost, but it's only for one matrix instead of making 4 (for 2D) and 6 (for 3D) each time the function is being called.

Reply:
It's a const function, so it is known at compile time, there is nothing more to do. It's more expensive to truncate and reconstruct a matrix than to just return a const. It's probably `inlined` this way of using `const` than your way of truncation. So, private member variable is not better and is more expensive.

Reply:
If you are editing a private member variable, it is no longer a `const` function.

Reply:
Alright. Let me do it